### PR TITLE
Added EditMessage, SendMessage, and DeleteMessage use cases and ChatR…

### PR DESCRIPTION
**Pull request by EricYH2134**
…epository and MessageRepository files where it saves to JSON persistence, along with interfaces and request models necessary to facilitate it in accordance to clean architecture

### CODE REVIEW

**Comment by nik-uoft** [comment]
Great work, Eric! We still need to add test coverage for these use cases!

**Comment by EricYH2134** [comment]
Added test cases, fixed some bugs, and changed some code to remove some IntelliJ warnings

**Comment by EricYH2134** [comment]
Closed because I realized that my commits were not attributed to my GitHub as git config on my computer did not have an email. So I closed this and recommitted with correct git settings in a new pull request so GitHub would attribute these commits to me. However, older commits under author "Eric" have not changed, but those are still my commits

**Review by rryt000 — COMMENTED** [review]
Overall, the Chat repository and use cases are looking great. As far as my knowledge of clean architecture goes, all the elements such as the Presenter, Gateway, Output and Input Boundaries, and response models are present. One thing to pay attention to is perhaps complete the overriden methods in the ChatPresenter class, and consider catching all the exceptions thrown (this may entail creating custom exceptions). We should also combine our repositories together, since they overlap in quite a few aspects. Other than that 👍

File: src/main/java/chat/ChatRepository.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    private int numChats;
+
+    public ChatRepository(String filePath) throws IOException {
+        this.filePath = filePath;
+        File JSONFile = new File(filePath);
+        this.numChats = 0;
+        if (JSONFile.length() > 0) {
+            FileReader reader = new FileReader(JSONFile);
+            Gson gson = new GsonBuilder().create();
+            chats = gson.fromJson(reader, new TypeToken<HashMap<Integer, ChatRepoRequestModel>>() {}.getType());
```
Comment (nik-uoft) [inline src/main/java/chat/ChatRepository.java:30]: nit: let's add `this` for `chats` to disambiguate variable references.

File: src/main/java/chat/ChatResponseModel.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    private String content;
+
+    private int author;
+
+    private int receiver;
+
+    private Date sendTime;
+
+    public ChatResponseModel(String content, int author, int receiver, Date sendTime) {
```
Comment (nik-uoft) [inline src/main/java/chat/ChatResponseModel.java:15]: nit: for the response model let's just use a single pair of getter and setter.

**Pull request closed without merging**